bayes_opt/target_space.py

```diff
@@ -209,11 +297,17 @@ def register(self, params, target, constraint_value=None):
         if x in self:
             if self._allow_duplicate_points:
                 self.n_duplicate_points = self.n_duplicate_points + 1
-                print(f'{Colours.RED}Data point {x} is not unique. {self.n_duplicate_points} duplicates registered.'
-                              f' Continuing ...{Colours.END}')
+
+                print(Fore.RED + f'Data point {x} is not unique. {self.n_duplicate_points}'
+                              ' duplicates registered. Continuing ...')
             else:
-                raise NotUniqueError(f'Data point {x} is not unique. You can set "allow_duplicate_points=True" to '
-                                     f'avoid this error')
+                raise NotUniqueError(f'Data point {x} is not unique. You can set'
+                                     ' "allow_duplicate_points=True" to avoid this error')
+
+        # if x is not within the bounds of the parameter space, warn the user
+        if self._bounds is not None:
+            if not np.all((self._bounds[:, 0] <= x) & (x <= self._bounds[:, 1])):
+                warn(f'\nData point {x} is outside the bounds of the parameter space. ', stacklevel=2)
```

Comment:
lines `307-310` appear to be duplicates of `312-316`

Reply:
My bad, this must have happened when trying to fix a merge conflict.